core/observe.c

```diff
@@ -925,12 +927,11 @@ int lwm2m_observe(lwm2m_context_t * contextP,
     transactionP = transaction_new(clientP->sessionH, COAP_GET, clientP->altPath, uriP, contextP->nextMID++, 4, token);
     if (transactionP == NULL)
     {
+        observationP->clientP->observationList = (lwm2m_observation_t *)LWM2M_LIST_RM(observationP->clientP->observationList, observationP->id, NULL);
```

Comment:
I'm not sure the logic is quite right here. An existing observation can be removed because the transaction to send the observe again can't be created. The client could still be sending notifies, but the server can no longer process them.